src/main/scala/io/dronekit/CouchbaseStreamsWrapper.scala

```diff
@@ -224,6 +229,17 @@ class CouchbaseStreamsWrapper(host: String, bucketName: String, password: String
     convertToEntity[T](replaceObservable)
   }
 
+  def insertDocumentWithBackoff(bucket: Bucket,  document: RawJsonDocument): Observable[RawJsonDocument] = {
+    for (i <- 1 to CouchbaseStreamsWrapper.temporaryFailureMaxRetries) {
+      try {
+        return bucket.async().insert(document)
+      } catch {
+        case ex: TemporaryFailureException => Thread.sleep(i * CouchbaseStreamsWrapper.temporaryFailureRetryMs)
```

Comment:
I think this function is otherwise async (Observable is like a Future), so I don't think we want to block here. I'm also not sure if the TemporaryFailureException is thrown synchronously, or in the onError notification of the Observable, in which case it wouldn't be caught by this.

If you scroll down a bit on https://developer.couchbase.com/documentation/server/4.0/sdks/java-2.2/documents-bulk.html#story-h2-5, there's a RetryBuilder class that might be what we want.

Reply:
Rewrote this to use the Retry Builder API